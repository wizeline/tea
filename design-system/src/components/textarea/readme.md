# wz-textarea



<!-- Auto Generated Below -->


## Properties

#### placeholder

string


#### value

string


## Attributes

#### placeholder

string


#### value

string



----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
