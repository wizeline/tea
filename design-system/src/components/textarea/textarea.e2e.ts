import { newE2EPage } from '@stencil/core/testing';

describe('wz-textarea', () => {
  it('renders textarea component', async () => {
    const page = await newE2EPage();
    await page.setContent(
      "<wz-textarea value='Hello, my name is Marty McFly'></wz-textarea>",
    );
    const textArea = await page.find('wz-textarea');
    const value = await textArea.getProperty('value');
    expect(value).toEqual('Hello, my name is Marty McFly');
  });
});
