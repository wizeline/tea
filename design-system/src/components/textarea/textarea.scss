wz-textarea {

  .error textarea {
    border-color: var(--negative-base);
    &:focus-within {
      border-color: var(--negative-base);
    }
  }

  wz-error {
    display: block;
    height: 16px;
  }

  textarea {
    padding: 8px 16px;
    border: none;
    background-color: var(--surface-base);
    border: solid 1px;
    border-color: var(--ink-light);
    font-size: 14px;
    resize: none;
    outline: none !important;
    width: 100%;

    &:focus {
      border-color:  var(--primary-light);
    }
  }

}
