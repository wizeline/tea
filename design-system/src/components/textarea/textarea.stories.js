import { storiesOf } from '@storybook/html';

storiesOf('Text Area', module).add(
  'default',
  () =>
    '<div style="width: 200px">' +
    '<wz-textarea value="Default Text area" oninput="(function(e) { console.log(this.event.target.value) })()"></wz-textarea>' +
    '</div>',
);
