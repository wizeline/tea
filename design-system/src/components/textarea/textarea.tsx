import { Component, Prop } from '@stencil/core';

@Component({
  tag: 'wz-textarea',
  styleUrl: 'textarea.scss',
  shadow: false,
})
export class TextArea {
  @Prop({ mutable: true })
  value: string;
  @Prop()
  name: string;
  @Prop()
  placeholder: string;
  @Prop()
  error: string;

  render() {
    const DEFAULT_ROWS = 3;

    return (
      <wz-form-error error={this.error}>
        <textarea
          name={this.name}
          placeholder={this.placeholder}
          rows={DEFAULT_ROWS}
          value={this.value}
        />
      </wz-form-error>
    );
  }
}
