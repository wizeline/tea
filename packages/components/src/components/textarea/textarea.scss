wz-textarea {
  .textarea-container {
    background-color: var(--ink-clear);
    border: solid 1px;
    border-color:  var(--neutral-base);

    &:focus-within {
      border-color:  var(--primary-light);
    }

    textarea {
      padding: 8px 16px;
      border: none;
      font-size: 14px;
      resize: none;
      outline: none !important;
      width: 100%;
      overflow: hidden;
    }
  }

}
