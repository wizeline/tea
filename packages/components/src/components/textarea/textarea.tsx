import { Component, Prop } from '@stencil/core';

@Component({
  tag: 'wz-textarea',
  styleUrl: 'textarea.scss',
  shadow: false,
})
export class TextArea {

  @Prop({ mutable: true }) value: string;

  render() {
    const DEFAULT_ROWS = 3;

    return (
      <div class="textarea-container">
        <textarea rows={DEFAULT_ROWS} value={this.value} />
      </div>
    );
  }
}
