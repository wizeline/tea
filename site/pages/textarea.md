### Default
```html
showSource: true
---
<wz-textarea placeholder="Write something" />
```

### With Error

```html
showSource: true
---
<wz-textarea
  placeholder="Write something"
  error="Please select at least one roadmap"
/>
```

### Disabled

```html
showSource: true
---
<wz-textarea
  disabled
  placeholder="Cannot Write"
/>
```
