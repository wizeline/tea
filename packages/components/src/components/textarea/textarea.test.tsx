import { TextArea } from './textarea';

import { TestWindow } from '@stencil/core/testing';

describe('wz-textarea', () => {
  it('renders value', async () => {
    const window = new TestWindow();
    const element = await window.load({
      components: [TextArea],
      html: "<wz-textarea value='Hello, my name is Marty McFly'></wz-textarea>",
    });
    const textarea = element.querySelector('textarea');

    expect(textarea.value).toEqual('Hello, my name is Marty McFly');
    expect(element).toMatchSnapshot();
  });
});
