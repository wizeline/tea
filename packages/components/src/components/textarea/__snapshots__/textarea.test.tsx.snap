// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wz-textarea renders value 1`] = `
<wz-textarea
  class="hydrated"
  value="Hello, my name is Marty McFly"
>
  
  <div
    class="textarea-container"
  >
    <textarea
      rows="3"
    />
  </div>
</wz-textarea>
`;
